src/features/groups/list/group-details/index.tsx

```diff
@@ -0,0 +1,184 @@
+
+import { memo, useCallback, useEffect, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { Row } from '../../../../components/Containers';
+import { ITransactionGroup } from '../../types';
+import {
+  AddMembersWrapper,
+  ContentWrapper,
+  GroupCurrencyText,
+  GroupDetailsBackground,
+  GroupDetailsWrapper,
+  GroupInfoWrapper,
+  MainInfoWrapper,
+  SaveButtonWrapper
+} from './index.styled';
+import addAvatar from '../../../../assets/images/add-avatar.png';
+import arrowRight from '../../../../assets/images/arrow-right.png';
+import Select from '../../../../components/Select';
+import { TextUnderline } from '../../../../components/Text';
+import currencies from '../../../../constants/currencies';
+import { useCreateTransactionGroupMutation, useLazyGetTransactionGroupByIdQuery, useUpdateTransactionGroupMutation } from '../../api';
+import Button from '../../../../components/Button';
+import { requiredValidator } from '../../../auth/validation';
+import Input from '../../../../components/Input';
+import GroupMembers from '../group-members';
+
+interface IValidation {
+  name: boolean;
+}
+
+export interface IGroupDetailsProps {
+  groupId?: string | null;
+  onSaved: (group: ITransactionGroup) => void;
+  isAddGroupMembersMode: boolean;
+  onGroupMembersModeChange: (value: boolean) => void;
+}
+
+const GroupDetails = ({ groupId, onSaved, isAddGroupMembersMode, onGroupMembersModeChange }: IGroupDetailsProps) => {
+  const { t } = useTranslation('groups');
+
+  const [getGroupById] = useLazyGetTransactionGroupByIdQuery();
+  const [createGroup, { data: createdGroup }] = useCreateTransactionGroupMutation();
+  const [updateGroup, { data: updatedGroup }] = useUpdateTransactionGroupMutation();
+
+  const [form, setForm] = useState<ITransactionGroup>({
+    id: '',
+    name: '',
+    currencyCode: currencies[0].value,
+    participants: [],
+    transactions: [],
+    ownerId: ''
```

Comment:
Why don't you prefill data for this group from the groups list if it is editing?
As I remember you only need to get participants from `useLazyGetTransactionGroupByIdQuery `

Also I think we need to move `participants` and `transactions` outside of the `form` state.
Because we can't edit transactions at all, and can't edit participants sending new array.
So it will be good to separate it. It allows us to prefill data from the groups list end edit this data immediately without waiting for getGroupById request.